### PR TITLE
Fix and optimize query profiling

### PR DESCRIPTION
**Pull request by Zoxc**
r? @michaelwoerister 

cc @wesleywiser

### CODE REVIEW

**Comment by Zoxc** [comment]
@rust-timer build e12bb21de0cda2992dd77eb54755b113dea148f5

File: src/librustc/util/profiling.rs [hunk 1]
```diff
@@ -178,14 +180,14 @@ impl SelfProfiler {
         self.timer_stack.push(category);
     }
 
-    pub fn record_query(&mut self, category: ProfileCategory) {
-        let (hits, total) = *self.data.query_counts.get(category);
-        self.data.query_counts.set(category, (hits, total + 1));
+    pub fn record_queries(&mut self, category: ProfileCategory, count: usize) {
+        let (hits, computed) = *self.data.query_counts.get(category);
+        self.data.query_counts.set(category, (hits, computed + count as u64));
```
Comment (michaelwoerister) [inline src/librustc/util/profiling.rs]: An issue for another PR: This looks like we could get missed updates here (and below) with parallel queries.

**Review by michaelwoerister — COMMENTED** [review]
Thanks, @Zoxc! Good idea to record query counts in bulk.
r=me with the functions renamed.

**Pull request merged**